sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -94,10 +95,11 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]]
    * All Attributes that appear in expressions from this operator.  Note that this set does not
    * include attributes that are implicitly referenced by being passed through to the output tuple.
    */
-  @transient
```

Comment:
Sorry for only spotting this now, but:

Previously we would recompute these attribute references if they were requested on a QueryPlan that had been serialized and shipped to executors, whereas after this PR's change we'll use attributes sent from the driver.

Are there any potentially adverse side effects of this, other than serialized size? I vaguely recall that there was some attribute JVM ID reference stuff somewhere, but I'm unsure offhand if that's a factor here.

I mention this because the `@transient` was added in https://github.com/apache/spark/pull/24866/commits/4483331586090dc5ddc79c646e1c18eb1aa0742c and I'm wondering if it was somehow load bearing or whether it's just a perf. optimization.

Reply:
good catch!
according to the description of https://github.com/apache/spark/pull/24866 , it mainly affect the planning time.

Maybe we can add `@transient` to `_references`

Reply:
shall we mark `Lazy#value` as `@transient`?

Reply:
Making `Lazy#value` a `@transient` field is an interesting proposal and seems promising.

If we do this, I think we should consider changing the name to `TransientLazy` or `LazyTransient` for the sake of making the transient aspect more salient to a reader / reviewer.

Reply:
+1 for `TransientLazy`